### DISCUSSION TITLE
Spell check in different languages

### DISCUSSION THREAD

**Discussion by peha2407** [SafeExamBrowser/seb-mac#334] [General]
Hi all, 
Our teachers ask for spell checkers in different languages (e.g. German, French, Italian, English) for their exams (essays). Does this possibility already exist or can you implement it?
Regards,
Peter

**Comment by danschlet** [comment]
In the Mac version of Safe Exam Browser, you just configure spell checking languages in macOS settings.

**Reply by peha2407** [reply]
Thank you for your answer. I have tried it, but the "Correct text" function does not work in the editor. We work with a special program that automatically creates a configuration. Maybe that is the problem. I'll have to have a look at this with the developer, I think.

**Comment by moehzhaw** [comment]
Hi 
We have the same or a similar issue, see our setting below.
= Can this be solved with ...
- ... a **configuration** (SEB config tool or macOS setting) or 
- ... is there a **difference** between Windows and macOS or 
- ... is Spell Checker **not possible any more** when 3.2.3 or older is in use?

**Result of my testings:**
- macOS: ❌Spell Checker does **not** work in a Moodle-Quiz
- Windows: ✅no problem, i.e. Spell Checker works

**We work with:**
- Moodle 4.3.8 
- with Atto-Editor
- Moodle Activity: Quiz (Essay question type)
- As "en" is set as language in the course, I changed the language in the macOS settings to "british english"

**SEB-Versions**
- macOS: 3.4.0 (According to the tutor, Spell Checker works with versions 3.2.2 or older.)
- Windows: 3.8.0

**Reply by danschlet** [reply]
Unfortunately Apple doesn't provide an API to safely disable spell checking in the current (modern) WKWebView browser engine after an application was initialized. Therefore I had to remove support for the macOS spell checker completely in [SEB for macOS 3.3.3](https://safeexambrowser.org/macosx/mac_release_notes_en.html) (as in a majority of  use cases spell checking cannot be active during an exam).

I don't know if there is some javascript-based spell checker in some web editors which works independently from the browser engine (operating system) spell checker. I assume that should exist, but maybe not as a Moodle plugin.

I reported this issue to Apple, but if they solve it in a future macOS release, it will only be available for macOS clients running on that or a newer macOS version.
